code/controllers/subsystem/minor_mapping.dm

```diff
@@ -16,26 +16,20 @@ SUBSYSTEM_DEF(minor_mapping)
 /datum/controller/subsystem/minor_mapping/proc/trigger_migration(num_mice=10)
 	var/list/exposed_wires = find_exposed_wires()
 
-	var/mob/living/basic/mouse/mouse
 	var/turf/open/proposed_turf
-
-
 	while((num_mice > 0) && exposed_wires.len)
 		proposed_turf = pick_n_take(exposed_wires)
-
-		if(!istype(proposed_turf))
+		if (!istype(proposed_turf) || \
+			!proposed_turf.air.has_gas(/datum/gas/oxygen, 5) || \
+			proposed_turf.temperature > NPC_DEFAULT_MAX_TEMP || \
+			proposed_turf.temperature < NPC_DEFAULT_MIN_TEMP)
 			continue
 
-		if(prob(PROB_MOUSE_SPAWN))
-			if(!mouse)
-				mouse = new(proposed_turf)
-			else
-				mouse.forceMove(proposed_turf)
+		num_mice -= 1
```

Comment:
does a decrement operator not work here like `num_mice--` or is that just not trustable for whatever reason (i know you just transplanted the code from one place to here but i'm just curious why this is)

Reply:
I don't even know why the code was originally written in the way it was, seemed a pretty roundabout way to do it. If I had noticed this I would have changed it.

Reply:
that's fine just wondering if you noticed it and didn't change it deliberately